### ISSUE TITLE
pdfsizeopt.main.FilterError: Ghostscript decompression with filter '/A85' failed: (...)

### ISSUE THREAD

**Issue by rbrito** [pts/pdfsizeopt#47]
Dear @pts,

I just stumbled upon a file here that I suspect that is malformed but, if preprocessed with, way GS or with qpdf, gets successfully optimized by pdfsizeopt.

OTOH, if it is not preprocessed, it gives me a stacktrace like this:

```
(...)
info: will optimize image XObject 1428; orig width=1860 height=2760 colorspace=/DeviceGray bpc=1 inv=False filter=/CCITTFaxDecode dp=1 size=42414 gs_device=pngmono
info: will optimize image XObject 1431; orig width=1860 height=2760 colorspace=/DeviceGray bpc=1 inv=False filter=/CCITTFaxDecode dp=1 size=43220 gs_device=pngmono
info: will optimize image XObject 1434; orig width=1860 height=2760 colorspace=/DeviceGray bpc=1 inv=False filter=/CCITTFaxDecode dp=1 size=41308 gs_device=pngmono
info: will optimize image XObject 1437; orig width=1860 height=2760 colorspace=/DeviceGray bpc=1 inv=False filter=/CCITTFaxDecode dp=1 size=41785 gs_device=pngmono
info: will optimize image XObject 1440; orig width=1860 height=2760 colorspace=/DeviceGray bpc=1 inv=False filter=/CCITTFaxDecode dp=1 size=40800 gs_device=pngmono
info: will optimize image XObject 1443; orig width=1860 height=2760 colorspace=/DeviceGray bpc=1 inv=False filter=/CCITTFaxDecode dp=1 size=43269 gs_device=pngmono
info: will optimize image XObject 1446; orig width=1860 height=2760 colorspace=/DeviceGray bpc=1 inv=False filter=/CCITTFaxDecode dp=1 size=30509 gs_device=pngmono
info: will optimize image XObject 3202; orig width=1860 height=2760 colorspace=/DeviceGray bpc=1 inv=False filter=/CCITTFaxDecode dp=1 size=26475 gs_device=pngmono
info: using Ghostscript /usr/bin/gs: GPL Ghostscript 9.21 (2017-03-16)
info: decompressing 62 bytes with Ghostscript /Filter/A85
GPL Ghostscript 9.21: Unrecoverable error, exit code 1
Traceback (most recent call last):
  File "/home/rbrito/Downloads/pdfsizeopt/pdfsizeopt", line 41, in <module>
    sys.exit(main.main(sys.argv, script_dir=script_dir))
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 9084, in main
    pdf.OptimizeImages(img_cmd_patterns=img_cmd_patterns)
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 6886, in OptimizeImages
    obj.Get('ColorSpace'), objs=self.objs, do_strings=True)
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 3480, in ResolveReferences
    data = cls.PDF_REF_RE.sub(Replacement, data)
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 3467, in Replacement
    return obj.SerializePdfStringSafe(obj.GetUncompressedStream(objs=objs))
  File "/home/rbrito/Downloads/pdfsizeopt/lib/pdfsizeopt/main.py", line 3389, in GetUncompressedStream
    (filter_value, gs_defilter_cmd, data))
pdfsizeopt.main.FilterError: Ghostscript decompression with filter '/A85' failed: gs -dNODISPLAY -sINFN=psotmp.16222.filter.tmp.bin -q -P- -c '/i INFN(r)file<</CloseSource true /Intent 2/Filter /A85>>/ReusableStreamDecode filter def /o(%stdout)(w)file def/s 4096 string def {i s readstring exch o exch writestring not{exit}if}loop o closefile quit' ('Error: /rangecheck in --.reusablestreamdecode--\nOperand stack:\n   i   --nostringval--   --dict:3/3(L)--   --nostringval--   --dict:3/3(L)--   --dict:3/3(L)--\nExecution stack:\n   %interp_exit   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   --nostringval--   --nostringval--   false   1   %stopped_push   .runexec2   --nostringval--   --nostringval--   --nostringval--   2   %stopped_push   --nostringval--   1862   4   3   %oparray_pop   1973   3   3   %oparray_pop   --nostringval--\nDictionary stack:\n   --dict:1210/1684(ro)(G)--   --dict:0/20(G)--   --dict:78/200(L)--\nCurrent allocation mode is local\n')

```

I'm sending you a link to the file by email.

Thanks,

Rogério.

**Comment by pts** [comment]
Thank you for reporting this! This is a duplicate of https://github.com/pts/pdfsizeopt/issues/23, le'ts continue there.

**Comment by pts** [comment]
Fixed in f77bf784e5a22f4804218ecdf6886ba5ec5e256e.

**Comment by rbrito** [comment]
Super thanks!
